api/token.go

```diff
@@ -245,12 +246,20 @@ func (a *API) issueRefreshToken(ctx context.Context, conn *storage.Connection, u
 func (a *API) setCookieToken(config *conf.Configuration, tokenString string, session bool, w http.ResponseWriter) error {
 	exp := time.Second * time.Duration(config.Cookie.Duration)
 	cookie := &http.Cookie{
+		Domain:   config.SiteURL,
 		Name:     config.Cookie.Key,
 		Value:    tokenString,
 		Secure:   true,
```

Comment:
We might have to check if we're in a secure environment, e.g. if SiteURL is in `['localhost', '127.0.0.1']` then `Secure` needs to be `false`.

Reply:
@thorwebdev can you pls describe why this is important?

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies: 
"A cookie with the Secure attribute is sent to the server only with an encrypted request over the HTTPS protocol, never with unsecured HTTP (except on localhost), and therefore can't easily be accessed by a man-in-the-middle attacker. Insecure sites (with http: in the URL) can't set cookies with the Secure attribute."

But I guess this means we're fine as long as the gotrue server is serving over https?